From: Liran Piade <liranpiade@gmail.com>
Date: Mon, 5 Jul 2021 14:18:42 +0000
Subject: Added ELAN 0c6e (from ROG Flow X13)

---
 data/autosuspend.hwdb    | 1 +
 libfprint/drivers/elan.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/data/autosuspend.hwdb b/data/autosuspend.hwdb
index afc86f4..b9ec5da 100644
--- a/data/autosuspend.hwdb
+++ b/data/autosuspend.hwdb
@@ -123,6 +123,7 @@ usb:v04F3p0C42*
 usb:v04F3p0C4D*
 usb:v04F3p0C4F*
 usb:v04F3p0C63*
+usb:v04F3p0C6E*
  ID_AUTOSUSPEND=1
 
 # Supported by libfprint driver etes603
diff --git a/libfprint/drivers/elan.h b/libfprint/drivers/elan.h
index 00afe79..e14977e 100644
--- a/libfprint/drivers/elan.h
+++ b/libfprint/drivers/elan.h
@@ -218,6 +218,7 @@ static const FpIdEntry elan_id_table[] = {
   {.vid = ELAN_VEND_ID,  .pid = 0x0c4d, .driver_data = ELAN_ALL_DEV},
   {.vid = ELAN_VEND_ID,  .pid = 0x0c4f, .driver_data = ELAN_ALL_DEV},
   {.vid = ELAN_VEND_ID,  .pid = 0x0c63, .driver_data = ELAN_ALL_DEV},
+  {.vid = ELAN_VEND_ID,  .pid = 0x0c6e, .driver_data = ELAN_ALL_DEV},
   {.vid = 0,  .pid = 0,  .driver_data = 0},
 };
 
