From: mbv06 <mbv06.dev@gmail.com>
Date: Tue, 27 Sep 2022 08:59:49 +0000
Subject: Update autosuspend.hwdb

---
 data/autosuspend.hwdb | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/autosuspend.hwdb b/data/autosuspend.hwdb
index d8127c3..ca2435a 100644
--- a/data/autosuspend.hwdb
+++ b/data/autosuspend.hwdb
@@ -212,6 +212,7 @@ usb:v147Ep1001*
 
 # Supported by libfprint driver upektc
 usb:v0483p2015*
+usb:v0483p2017*
 usb:v147Ep3001*
  ID_AUTOSUSPEND=1
  ID_PERSIST=0
